Consul.AspNetCore/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Consul.AspNetCore
+{
+	public static class ServiceCollectionExtensions
+	{
+		public static IServiceCollection AddConsul(this IServiceCollection services)
+		{
+			return services.AddConsul(options => {});
+		}
+
+		public static IServiceCollection AddConsul(
+			this IServiceCollection services,
+			Action<ConsulClientConfiguration> options)
+		{
+			services.TryAddSingleton<IConsulClient>(sp => new ConsulClient(options));
+
+			return services;
+		}
+
+		public static IServiceCollection AddConsulRegistration(
```

Comment:
I find the method name a bit confusing. IMO 'AddConsulServiceRegistration' would be more descriptive and would allow extending the class later with check registration.